src/Microsoft.AspNet.Mvc.Razor/VirtualPathRazorPageFactory.cs

```diff
@@ -14,40 +13,23 @@ namespace Microsoft.AspNet.Mvc.Razor
     /// </summary>
     public class VirtualPathRazorPageFactory : IRazorPageFactory
     {
-        private readonly IServiceProvider _serviceProvider;
+        private readonly Func<RelativeFileInfo, CompilationResult> _compileDelegate;
```

Comment:
None in particular. Seemed sort of intuitive since we don't need access to `IRazorCompilationService` later on.